### DISCUSSION TITLE
Invalid Character Error

### DISCUSSION THREAD

**Discussion by ramseyjiang** [spatie/array-to-xml#179] [Q&A]
DOMException 

  Invalid Character Error

  at vendor/spatie/array-to-xml/src/ArrayToXml.php:193

**Comment by webtamizhan** [comment]
Can you please share sample array here.

I have one solution for this issue.

The same problem happened to me because of isArrayAllKeySequential method. It's happen because of mixed keys data types in an array. i.e combination of int & string. So keep one value and manually add the element attribute or new element using xpath.

`$dom = new \DOMDocument();`
       ` $dom->load($file);`
       ` $xp = new \DOMXPath($dom);`
       ` $res = $xp->query("YOUR XPATH QUERY HERE");`
       ` if($res->length > 0) {`
           ` foreach ($res as $index => $re) {`
                `$res->item($index)->setAttribute('SEGMENT','1');`
           ` }`
            `$dom->save($file);`
       ` }`

**Comment by ArduanovDanil** [comment]
I have same error with example from docs

```php
$array = [
    '__custom:ns\\:custom-key:1' => [
        'name' => 'Vladimir',
        'nickname' => 'greeflas',
    ],
    '__custom:ns\\:custom-key:2' => [
        'name' => 'Marina',
        'nickname' => 'estacet',
        'tags' => [
            '__custom:ns\\:tag:1' => 'first-tag',
            '__custom:ns\\:tag:2' => 'second-tag',
        ]
    ],
];

$result = ArrayToXml::convert($array);
```
I whant to use xml prefix like this

```
<ns:custom-key>
</ns:custom-key>
```

but I can't

**Comment by wmdkdkussin** [comment]
FYI: We had the same issue which was caused by an empty array key:
```php
$aData = explode('=', '=braun');
```
As the array key is used as XML node the node was invaild as it was empty.
